File: jafu/src/test/java/org/springframework/fu/jafu/webflux/MustacheDslTests.java
package org.springframework.fu.jafu.webflux;

import static org.springframework.fu.jafu.Jafu.reactiveWebApplication;
import static org.springframework.fu.jafu.webflux.WebFluxServerDsl.webFlux;
import static org.springframework.web.reactive.function.server.ServerResponse.ok;

import java.util.Collections;

import org.junit.jupiter.api.Test;

import org.springframework.test.web.reactive.server.WebTestClient;

public class MustacheDslTests {

	@Test
	void createAndRequestAMustacheView() {
		var app = reactiveWebApplication(a -> a.enable(webFlux(s -> s.port(0).mustache().router(r -> r.GET("/view", request -> ok().render("template", Collections.singletonMap("name", "world")))))));

		var context = app.run();
		var port = context.getEnvironment().getProperty("local.server.port");
		var client = WebTestClient.bindToServer().baseUrl("http://127.0.0.1:" + port).build();
		client.get().uri("/view").exchange()
				.expectStatus().is2xxSuccessful()
				.expectBody(String.class)
				.isEqualTo("Hello world!");
		context.close();
	}

}

File: kofu/src/test/kotlin/org/springframework/fu/kofu/web/MustacheDslTests.kt
/*
 * Copyright 2002-2018 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.springframework.fu.kofu.web

import org.junit.jupiter.api.Test
import org.springframework.fu.kofu.application
import org.springframework.test.web.reactive.server.WebTestClient
import org.springframework.test.web.reactive.server.expectBody
import org.springframework.web.reactive.function.server.router
import reactor.netty.http.HttpResources

/**
 * @author Sebastien Deleuze
 */
class MustacheDslTests {

	@Test
	fun `Create and request a Mustache view`() {
		val router = router {
			GET("/view") { ok().render("template", mapOf("name" to "world")) }
		}
		val app = application {
			server {
				mustache()
				import(router)
			}
		}
		app.run()
		val client = WebTestClient.bindToServer().baseUrl("https://0.0.0.0:8080").build()
		client.get().uri("/view").exchange()
			.expectStatus().is2xxSuccessful
			.expectBody<String>()
			.isEqualTo("Hello world!")
		app.stop()
	}

}